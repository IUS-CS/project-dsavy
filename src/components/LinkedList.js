import React, {useState} from 'react';
import { useSpring, animated, Trail } from 'react-spring';
import Node from './List-Node';
import ListForm from './List-Form';

const LinkedList = () => {
    const [list, setList] = useState([]);

    const [key, setKey] = useState(1);
  
    const scrolling = useSpring({
      from: { transform: "translate(60%,0)" },
      to: { transform: "translate(-60%,0)" },
      config: { duration: 20000 },
      reset: true,
      //reverse: key % 2 == 0,
      onRest: () => {
        setKey(key + 1);
      }
    })
  
    return (
      <div>
        <div key={key} style={{paddingTop:120}}>
          <animated.div style={scrolling}><Node/></animated.div>
        </div>
        <ListForm list={list} onListChange={setList}/>
        <h2>{list}</h2>
      </div>
    )
  }
  
  export default LinkedList;