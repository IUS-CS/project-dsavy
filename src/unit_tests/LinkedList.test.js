import React from 'react';
import LinkedList from 'components/LinkedList';
import renderer from 'react-test-renderer';

test("Linked list page renders correctly", () => {
    const tree = renderer.create(<LinkedList/>).toJSON()
    expect(tree).toMatchSnapshot()
});