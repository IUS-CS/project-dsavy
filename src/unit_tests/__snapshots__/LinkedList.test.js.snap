// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Linked list page renders correctly 1`] = `
<div>
  <div>
    <div
      className="container-fluid"
      style={
        Object {
          "paddingTop": 60,
        }
      }
    >
      <div
        className="row"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 500,
            "justifyContent": "center",
            "paddingTop": 15,
          }
        }
      >
        Insert
      </div>
      <div
        className="row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <input
            className="my-1 mr-2 form-control"
            name="Value"
            placeholder="Value"
            type="text"
          />
          <label
            className="my-1 mr-2 form-label"
            htmlFor="inlineFormCustomSelectPref"
          >
            At Position
          </label>
          <select
            className="my-1 mr-sm-2 form-control"
            id="inlineFormCustomSelectPref"
            name="Place"
          >
            <option
              value="Tail"
            >
              Tail
            </option>
            <option
              value="0"
            >
              Head
            </option>
            <option
              value="1"
            >
              One
            </option>
            <option
              value="2"
            >
              Two
            </option>
            <option
              value="3"
            >
              Three
            </option>
          </select>
          <button
            className="my-1 btn btn-primary"
            disabled={false}
            type="submit"
          >
            Insert
          </button>
        </form>
      </div>
      <div
        className="row"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 500,
            "justifyContent": "center",
            "paddingTop": 15,
          }
        }
      >
        Remove
      </div>
      <div
        className="row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <label
            className="my-1 mr-2 form-label"
            htmlFor="inlineFormCustomSelectPref"
          >
            Element at Position
          </label>
          <select
            className="my-1 mr-sm-2 form-control"
            id="inlineFormCustomSelectPref"
            name="Place"
          >
            <option
              value="0"
            >
              Head
            </option>
            <option
              value="1"
            >
              One
            </option>
            <option
              value="2"
            >
              Two
            </option>
            <option
              value="3"
            >
              Three
            </option>
            <option
              value="Tail"
            >
              Tail
            </option>
          </select>
          <button
            className="my-1 btn btn-primary"
            disabled={false}
            style={Object {}}
            type="submit"
          >
            Remove
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "inline-flex",
        "paddingLeft": 45,
        "paddingTop": 20,
      }
    }
  />
</div>
`;

exports[`Linked list renders correctly 1`] = `
<div>
  <div>
    <div
      className="container-fluid"
      style={
        Object {
          "paddingTop": 60,
        }
      }
    >
      <div
        className="row"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 500,
            "justifyContent": "center",
            "paddingTop": 15,
          }
        }
      >
        Insert
      </div>
      <div
        className="row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <input
            className="my-1 mr-2 form-control"
            name="Value"
            placeholder="Value"
            type="text"
          />
          <label
            className="my-1 mr-2 form-label"
            htmlFor="inlineFormCustomSelectPref"
          >
            At Position
          </label>
          <select
            className="my-1 mr-sm-2 form-control"
            id="inlineFormCustomSelectPref"
            name="Place"
          >
            <option
              value="Tail"
            >
              Tail
            </option>
            <option
              value="0"
            >
              Head
            </option>
            <option
              value="1"
            >
              One
            </option>
            <option
              value="2"
            >
              Two
            </option>
            <option
              value="3"
            >
              Three
            </option>
          </select>
          <button
            className="my-1 btn btn-primary"
            disabled={false}
            type="submit"
          >
            Insert
          </button>
        </form>
      </div>
      <div
        className="row"
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 500,
            "justifyContent": "center",
            "paddingTop": 15,
          }
        }
      >
        Remove
      </div>
      <div
        className="row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <form
          className="form-inline"
          onSubmit={[Function]}
        >
          <label
            className="my-1 mr-2 form-label"
            htmlFor="inlineFormCustomSelectPref"
          >
            Element at Position
          </label>
          <select
            className="my-1 mr-sm-2 form-control"
            id="inlineFormCustomSelectPref"
            name="Place"
          >
            <option
              value="0"
            >
              Head
            </option>
            <option
              value="1"
            >
              One
            </option>
            <option
              value="2"
            >
              Two
            </option>
            <option
              value="3"
            >
              Three
            </option>
            <option
              value="Tail"
            >
              Tail
            </option>
          </select>
          <button
            className="my-1 btn btn-primary"
            disabled={false}
            style={Object {}}
            type="submit"
          >
            Remove
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    className="row"
    style={
      Object {
        "display": "inline-flex",
        "paddingLeft": 45,
        "paddingTop": 20,
      }
    }
  />
</div>
`;
